File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<int> a;
        for(int i = 0 ; i<s.size();i++)
        {
            if(s[i] == '(' || s[i] == '[' || s[i] == '{')
            a.push(s[i]);
            else
            {
                if(a.empty() == false)
                {
                    if(s[i] == ')' && a.top() == '(')
                    a.pop();
                    else if(s[i] == ']' && a.top() == '[')
                    a.pop();
                    else if(s[i] == '}' && a.top() == '{')
                    a.pop();
                    else
                    return false;
                }
                else
                return false;
            }
        }
        if(a.empty() == true)
        return true;
        return false;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        hash = {')':'(' , '}' : '{' , ']':'['}
        for i in s:
            if i in hash:
                if stack and stack[-1] == hash[i]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(i)
        return not stack
        